### PR TITLE
Neon Number in c

### PR DESCRIPTION
**Pull request by Madhumitamoi**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Yes I have been through the guidelines.)

### Description

Added Neon Number.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#2449

### CODE REVIEW

**Comment by Madhumitamoi** [comment]
@HarshCasper Please review.

File: C/math/Neon_number.c [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    if (sum == n)
+    {
+        printf("Neon number.\n");
+    }
+    else
+    {
+        printf("Not a neon number\n");
+    }
+    return 0;
+}
```
Comment (siddharth25pandey) [inline C/math/Neon_number.c]: As you are not returning anything, better to create a void function

File: C/math/Neon_number.c [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+int main()
+{
+    int n;
+    printf("Enter a number to check wheather it is neon or not- ");
```
Comment (siddharth25pandey) [inline C/math/Neon_number.c]: Change the sentence format and remove the *-* from the end.

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@Madhumitamoi Some modifications are required. Rest is fine.

**Comment by Madhumitamoi** [comment]
@HarshCasper Please review

**Review by siddharth25pandey — APPROVED** [review]
@Madhumitamoi Good to go. Able to merge !

File: C/math/Neon_number.c [hunk 3]
```diff
@@ -0,0 +1,37 @@
+// C Program to check whether a number is Neon or not. A neon number is a number where the sum of digits of square of the number is equal to the number.
```
Comment (HarshCasper) [inline C/math/Neon_number.c]: Modularize this content: Break it up into multiple lines and use multi-line comments

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Rest looks good to merge :+1:

**Comment by Madhumitamoi** [comment]
New Changes added.

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@Madhumitamoi Add space and time complexity.

**Comment by Madhumitamoi** [comment]
Space and time complexity added.

**Review by iamrajiv — APPROVED** [review]

**Comment by iamrajiv** [comment]
@all-contributors bot please add @Madhumitamoi for code

**Pull request merged**